src/java.base/share/classes/sun/security/jca/ProviderConfig.java

```diff
@@ -162,6 +171,12 @@ public String toString() {
     // com.sun.net.ssl.internal.ssl.Provider has been deprecated since JDK 9
     @SuppressWarnings("deprecation")
     synchronized Provider getProvider() {
+        // If provider is not allowed in restricted security mode, return without load.
+        if (RestrictedSecurityConfigurator.isEnabled()) {
+            if (!RestrictedSecurityProperties.getInstance().isProviderAllowed(provName)) {
+                return null;
+            }
+        }
```

Comment:
@keithc-ca 
In this ProviderConfig.java file, after line 355, there was a "p == null" check I added, but I already removed it. Because the not-allowed provider checks are added in the ServiceLoader.java before calling the iterator hasNext() method. So the provider p returned by the iter.next() won't be null.